src/erlcloud_cloudformation.erl

```diff
@@ -659,6 +725,137 @@ extract_account_limit(XmlNode) ->
             {value, "Value", optional_text}
         ], XmlNode).
 
-
-
+create_stack_input_to_params(Spec) ->
+    lists:flatten([
+        base_create_stack_input_params(Spec),
+        erlcloud_util:encode_object_list(
+            "Parameters",
+            cloudformation_parameters_fields(Spec#cloudformation_create_stack_input.parameters)),
+        erlcloud_util:encode_list("Capabilities", Spec#cloudformation_create_stack_input.capabilities),
+        erlcloud_util:encode_list("NotificationARNs", Spec#cloudformation_create_stack_input.notification_arns),
+        erlcloud_util:encode_list("ResourceTypes", Spec#cloudformation_create_stack_input.resource_types),
+        erlcloud_util:encode_object(
+            "RollbackConfiguration",
+            cloudformation_rollback_configuration_fields(Spec#cloudformation_create_stack_input.rollback_configuration)
+        ),
+        erlcloud_util:encode_object_list(
+            "Tags",
+            cloudformation_tags_fields(Spec#cloudformation_create_stack_input.tags)
+        )
+    ]).
+
+base_create_stack_input_params(Spec) ->
+    [
+        { "ClientRequestToken", Spec#cloudformation_create_stack_input.client_request_token },
+        { "DisableRollback", Spec#cloudformation_create_stack_input.disable_rollback },
+        { "EnableTerminationProtection", Spec#cloudformation_create_stack_input.enable_termination_protection },
+        { "OnFailure", Spec#cloudformation_create_stack_input.on_failure },
+        { "RoleARN", Spec#cloudformation_create_stack_input.role_arn },
+        { "StackName", Spec#cloudformation_create_stack_input.stack_name },
+        { "StackPolicyBody", Spec#cloudformation_create_stack_input.stack_policy_body },
+        { "StackPolicyURL", Spec#cloudformation_create_stack_input.stack_policy_url },
+        { "TemplateBody", Spec#cloudformation_create_stack_input.template_body },
+        { "TemplateURL", Spec#cloudformation_create_stack_input.template_url },
+        { "TimeoutInMinutes", Spec#cloudformation_create_stack_input.timeout_in_minutes }
+    ].
```

Comment:
Do we need to call `filter_undefined()` here as well?

Reply:
undefined get filtered out here https://github.com/erlcloud/erlcloud/blob/master/src/erlcloud_aws.erl#L1055 we just need to worry about filtering out nested fields like parameters, capabilities etc